wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/MachineInstanceProvider.java

```diff
@@ -51,8 +56,8 @@ Instance startService(String namespace,
                           boolean isDev,
                           String networkName,
                           CheServiceImpl service,
-                          LineConsumer machineLogger)
-            throws ServerException;
+                          LineConsumer machineLogger,
+                          List<? extends ServerConf> providedServers) throws ServerException;
```

Comment:
This class works with kinda docker terminology. It knows nothing about serversConf and other Che model staf. So it is better to move this code into CheEnvironmentEngine that should apply serverConf to CheServiceImpl to describe docker container. Then server conf can be recovered from docker container labels/exposed ports. You can find that logick in DockerInstanceRuntimeInfo class